pkgs/games/polymc/default.nix

```diff
@@ -23,34 +24,43 @@ in mkDerivation rec {
       --replace 'scanJavaDir("/usr/lib/jvm")' 'javas.append("${jdk}/lib/openjdk/bin/java")' \
       --replace 'scanJavaDir("/usr/lib32/jvm")' 'javas.append("${jdk8}/lib/openjdk/bin/java")'
 
+  '' + lib.optionalString (msaClientID != "") ''
     # add client ID
-    substituteInPlace notsecrets/Secrets.cpp \
-      --replace 'QString MSAClientID = "";' 'QString MSAClientID = "${msaClientID}";'
+    substituteInPlace CMakeLists.txt \
+      --replace '17b47edd-c884-4997-926d-9e7f9a6b4647' '${msaClientID}'
   '';
 
-  cmakeFlags = [ "-DMultiMC_LAYOUT=lin-system" ];
+  cmakeFlags = [ "-DLauncher_LAYOUT=lin-system" ];
+
+  desktopItem = makeDesktopItem {
+    name = "polymc";
+    desktopName = "PolyMC";
+    genericName = "Minecraft Launcher";
+    comment = "Free, open source launcher and instance manager for Minecraft.";
+    icon = "launcher";
+    exec = "polymc";
+    categories = "Game";
+    terminal = "false";
+  };
 
   postInstall = ''
-    install -Dm644 ../launcher/resources/multimc/scalable/multimc.svg $out/share/pixmaps/multimc.svg
-    install -Dm755 ../launcher/package/linux/multimc.desktop $out/share/applications/multimc.desktop
+    install -Dm644 ../launcher/resources/multimc/scalable/launcher.svg $out/share/pixmaps/polymc.svg
+    install -Dm755 ${desktopItem}/share/applications/polymc.desktop $out/share/applications/polymc.desktop
```

Comment:
Use the `DesktopItems` trio instead of a custom copy.

Reply:
done